apps/dg/components/graph/plots/univariate_plot_view.js

```diff
@@ -78,10 +78,20 @@ DG.UnivariatePlotView = DG.PlotView.extend(
       }.observes('*model.secondaryAxisPlace', 'xAxisView', 'yAxisView'),
 
       /**
-       * The secondaryAxisView needs to be told that its tick marks and labels are not to be centered in each cell.
-       * Though this is the default, if the prior plot was a dot chart, the axis will be stuck in centering mode.
+       * Prepare axes for univariate plot, including resetting plot-specific values.
        */
       setupAxes: function () {
+        // Review question: would these resets make more sense in DG.PlotView?
```

Comment:
Actually, after a more detailed look I decided not to move it since each branch of the hierarchy already has its own notion of what it means to setupAxes.

Reply:
I'm fine with whatever you decide here. The reason for the question is that some plots apply plot-specific properties to the axes. These resets are designed to clear such plot-specific properties from the axes when switching away from a plot that applied them. A better architecture would allow for an `unSetupAxes()` call to be made just before a plot went away so that each plot could be responsible for cleaning up after itself. With this approach, we're relying on the _next_ plot to clean up after the previous plot. This will work fine in the case of moving from one univariate plot to another, but there's no guarantee that the next plot will be a univariate plot, so it seems to me there may be scenarios in which the cleanup wouldn't occur. The advantage of doing cleanup in the base class (DG.PlotView) is that the next plot is guaranteed to be a derived class of DG.PlotView, but as you rightly point out the base class doesn't necessarily know enough to clean up properly.